### PR TITLE
feat: add group mapping option for group sync

### PR DESCRIPTION
**Pull request by coadler**
Allows mapping group IDs returned from an OIDC provider to a Coder group with a different name.

### CODE REVIEW

File: enterprise/coderd/userauth_test.go [hunk 1]
```diff
@@ -67,6 +67,51 @@ func TestUserOIDC(t *testing.T) {
...
 			require.Len(t, group.Members, 1)
 		})
+		t.Run("AssignsMapped", func(t *testing.T) {
+			t.Parallel()
+
+			ctx, _ := testutil.Context(t)
+			conf := coderdtest.NewOIDCConfig(t, "")
+
+			oidcGroupName := "pingpong"
+			coderGroupName := "bingbong"
```
Comment (kylecarbs) [inline enterprise/coderd/userauth_test.go:77]: classic

**Review by kylecarbs — APPROVED** [review]
I didn't review the docs, but code LGTM!

File: enterprise/coderd/userauth_test.go (hunk 1 above)
Comment (coadler) [reply enterprise/coderd/userauth_test.go:77]: classic

**Review by bpmct — APPROVED** [review]
Docs LGTM

**Pull request merged**